qt/python/mantidqt/widgets/algorithmselector/widget.py

```diff
@@ -252,7 +256,20 @@ def execute_algorithm(self):
         Send a signal to a subscriber to execute the selected algorithm
         """
         algorithm = self.get_selected_algorithm()
+        presets = {}
+        enabled = []
         if algorithm is not None:
+            if self.get_selected_workspace_fn:
+                selected_ws_names = self.get_selected_workspace_fn()
+                if selected_ws_names:
+                    selected = selected_ws_names[0]
```

Comment:
This line could be moved into the inner if-statement since `selected` is only used there.